projects/igniteui-angular/src/lib/grids/grid/grid-validation.service.ts

```diff
@@ -126,14 +126,14 @@ export class IgxGridValidationService {
      * @hidden
      * @internal
      */
-    public markAsTouched(rowId: any) {
+    public markAsTouched(rowId: any, field?: string) {
         const rowGroup = this.getFormGroup(rowId);
         if (!rowGroup) return;
         rowGroup.markAsTouched();
         for (const col of this.grid.columns) {
-            const field = col.field;
-            const control = rowGroup?.get(field);
-            control.markAsTouched();
+            if(!field || (field && col.field === field)) {
+                rowGroup?.get(col.field)?.markAsTouched();
+            }
```

Comment:
Mnor: You could easily short-circuit the loop if a field is passed to avoid looping all columns. I highly doubt it'll make much difference, but still. If you want to skip extending the code into if-else clause, you could loop over fields assigned in advance with a check though, like:
```ts
const fields = field ? [field] : this.grid.columns.map(x => x.field);
for (... fields)
```